zcash_client_backend/src/data_api/ll.rs

```diff
@@ -0,0 +1,554 @@
+    ///
+    /// In addition, for HD-derived addresses, the change-level key scope used to derive the
+    /// address is returned, so that the caller is able to determine whether any special handling
+    /// rules apply to the address for the purposes of preserving user privacy (by limiting address
+    /// linking, etc.).
+    #[cfg(feature = "transparent-inputs")]
+    #[allow(clippy::type_complexity)]
+    fn find_account_for_transparent_address(
+        &self,
+        address: &TransparentAddress,
+    ) -> Result<Option<(Self::AccountId, Option<TransparentKeyScope>)>, Self::Error>;
+
+    /// Detects the set of accounts that received transparent outputs corresponding to the provided
+    /// [`OutPoint`]s. This is used to determine which account(s) funded a given transaction.
+    ///
+    /// [`OutPoint`]: transparent::bundle::OutPoint
+    #[cfg(feature = "transparent-inputs")]
+    fn detect_accounts_transparent<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a transparent::bundle::OutPoint>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Detects the set of accounts that received Sapling outputs that, when spent, reveal(ed) the
+    /// given [`Nullifier`]s. This is used to determine which account(s) funded a given
+    /// transaction.
+    ///
+    /// [`Nullifier`]: sapling::Nullifier
+    fn detect_accounts_sapling<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a sapling::Nullifier>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Detects the set of accounts that received Orchard outputs that, when spent, reveal(ed) the
+    /// given [`Nullifier`]s. This is used to determine which account(s) funded a given
+    /// transaction.
+    ///
+    /// [`Nullifier`]: orchard::note::Nullifier
+    #[cfg(feature = "orchard")]
+    fn detect_accounts_orchard<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a orchard::note::Nullifier>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Get information about a transparent output controlled by the wallet.
+    ///
+    /// # Parameters
+    /// - `outpoint`: The identifier for the output to be retrieved.
+    /// - `spendable_as_of`: The target height of a transaction under construction that will spend the
+    ///   returned output. If this is `None`, no spendability checks are performed.
+    #[cfg(feature = "transparent-inputs")]
+    fn get_wallet_transparent_output(
+        &self,
+        outpoint: &OutPoint,
+        spendable_as_of: Option<TargetHeight>,
+    ) -> Result<Option<WalletUtxo>, Self::Error>;
+
+    /// Returns the vector of transactions in the wallet that spend the outputs of the referenced
+    /// transaction.
+    ///
+    /// This is used as part of [`wallet::store_decrypted_tx`] to allow downstream transactions'
+    /// fee amounts to be updated once the value of all their inputs are known.
+    fn get_spending_transactions(
```

Comment:
Document whether or not this API should return conflicted transactions that attempt to spend common outputs (particularly if none of them are mined or expired, which could occur in certain combinations of reorgs). I suspect we always need to return conflicts if none are mined, but maybe if one is mined we can exclude the rest? What is the most consistent response that trait implementers can follow?